### PR TITLE
Fix #127 - Updated Editor page with latest features

### LINKED ISSUE #127: latest PyEditor changes
There's quite a lot to unpack from this MR https://github.com/pyscript/pyscript/pull/2095 and we should have proper documentation once that lands as next release.

### PR DESCRIPTION
**Pull request by WebReflection**
This MR simply addresses its related issue, describing with words and examples what's new in our latest editor (soon to be released).

### CODE REVIEW

File: docs/user-guide/editor.md [hunk 1]
```diff
@@ -173,6 +173,47 @@ If a `setup` editor is present, that's the only PyEditor that needs a config.
...
+
+foreign.handleEvent = handle_event
+</script>
+```
+
+[This live example](https://agiammarchi.pyscriptapps.com/pyeditor-iot-example/latest/) that allows one to play with micro controllers running MicroPython shows how the editor can be used to execute code via serial USB connection.
+
+## Tab behavior
+
+We are currently trapping the `tab` key in a way that reflects what regular *IDE* do: the code is simply indented, as opposite of losing the focus every single time.
```
Comment (FabioRosado) [inline docs/user-guide/editor.md:210]: Do you want to uppercase this one too?

```suggestion
We are currently trapping the `Tab` key in a way that reflects what regular *IDE* do: the code is simply indented, as opposite of losing the focus every single time.
```

**Review by FabioRosado — APPROVED** [review]

**Comment by ntoll** [comment]
Hey hey :wave: - greetings from London. I've just pushed some language clean-ups and edits. Great work on this BTW.

Onwards! :rocket:

File: docs/user-guide/editor.md [hunk 2]
```diff
@@ -173,15 +173,17 @@ If a `setup` editor is present, that's the only PyEditor that needs a config.
...
-## Execution via keyboard
+## Run via keyboard
 
-Accordingly with the Operating system you are using, a combination of either `Ctrl-Enter`, `Cmd-Enter` or `Shift-Enter` would implicitly execute the code and show results within the `target` output node without ever needing to reach the *Run* button and lose the focus.
+Depending on your operating system, a combination of either `Ctrl-Enter`,
+`Cmd-Enter` or `Shift-Enter` will execute the code in the editor (no need to
+move the mouse to click the run button).
 
-## Execution Override
+## Execution override
```
Comment (WebReflection) [inline docs/user-guide/editor.md:182]: as you changed previous *Execution via keyword* to *Run* ... shouldn't this be then *Run override* too?

**Comment by WebReflection** [comment]
@ntoll thanks for the refinement, I am really bad at docs 😅 ... I have a minor *nit* though, because I think keeping the wording the same across these changes might be helpful ... I take *NO* as an answer, and I'd be happy to merge this whatever outcome there is 👍

File: docs/user-guide/editor.md (hunk 2 above)
Comment (ntoll) [reply docs/user-guide/editor.md]: Good point. Fixed.

**Comment by ntoll** [comment]
I'll stack upcoming documentation changes on top of this PR, just so everything relating to 2024-6-2 lands all at once. See: https://github.com/pyscript/docs/tree/2024-6-2

This PR should automatically close when the stacked branch is merged.

**Pull request merged**